lib/thrift/parser/resolver.ex

```diff
@@ -30,6 +30,16 @@ defmodule Thrift.Parser.Resolver do
       end
     end)
 
-    Map.merge(state, new_mappings)
+    existing_keys = Map.keys(state) |> MapSet.new
+    new_keys = Map.keys(new_mappings) |> MapSet.new
+    shared_keys = MapSet.intersection(existing_keys, new_keys) |> MapSet.to_list
+
+    case shared_keys do
+      [] ->
+        Map.merge(state, new_mappings)
+      [_ | _] ->
+        shared_keys_string = Enum.map_join(shared_keys, ", ", &to_string/1)
+        raise "Name collision: #{shared_keys_string}"
```

Comment:
It's got that, sorta. For example in the test the key is "dupes.Foo" because the thrift file is "dupes.thrift".